src/core/cgroup.c

```diff
@@ -817,9 +818,9 @@ int cgroup_log_xattr_apply(Unit *u, const char *cgroup_path) {
         if (!patterns)
                 return log_oom_debug();
 
-        memcpy_safe(patterns, allowed_patterns, allowed_patterns_len);
-        patterns[allowed_patterns_len] = '\xff';
-        memcpy_safe(&patterns[allowed_patterns_len + 1], denied_patterns, denied_patterns_len);
+        last = mempcpy_safe(patterns, allowed_patterns, allowed_patterns_len);
+        *last++ = '\xff';
```

Comment:
hmm, i have a hard time reading this, because off the top of my head i am not sure whether `*` or `++` bind stronger. maybe avoid the ambiguity and just write `*(last++) = …`